### PR TITLE
feat(dev): visualize network requests in the development environment on Chrome Devtools's network tab

### PR DESCRIPTION
**Pull request by GrinZero**
## Involved Issue / 该 PR 相关 Issue

NOROUTE

## Example for the Proposed Route(s) / 路由地址示例

```routes
NOROUTE
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [x] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

RSSHub 是一个非常依赖爬虫的项目，大部分 RSS 源都需要在插件中通过 got(ofetch) 爬取数据后进行加工。这个过程中，会发现调试过程并不显得非常棒，尤其是习惯了能通过 Chrome Devtools 可视化网络请求 payload、body 的开发者来说。

为了解决这个问题，增加开发体验，通过增加`node-network-devtools`包，这个包极大程度优化了网络调试过程的友好程度。

以下图为例，很容易从可视化的数据中发现某个插件存在部分损坏，而这个过程如果在控制台查看，detail 的大部分日志会被忽略。

Pasted Graphic

预期中还有的额外场景是～ got(ofetch) 遇到 302 请求的自动 follow 行为，不一定可以被及时注意到，而通过可视化的方式或许可以进行提前发现，进而处理成提前中断(当新地址无作用时)来提升性能。

这一版本，我对`node-network-devtools`做了特制优化，确保不会出现像上次的 PR 一样出现因不存在 Chrome 而导致程序阻塞的问题。同时，请忽略仅会出现一次的日志`MainProcess being already running`，可以忽略，因为`tsx`导致其重复运行两次，这实际上是一个警告。

相信我，在安装了`Chrome`的电脑上调试效果非常好～甚至 node 官方 v22.6.0 实验性的网络调试方案之所以能完成，也是因为我在其中的一些推进。

其次，我在 Follow 的社区找到了一些 RSSHub 贡献者做尝试，他们认为可视化的能力非常有价值！

请不要关闭这个 PR，遇到了任何问题请提交评论，我会进行处理！RSSHub 的主动网络请求之多，可以和`node-network-devtools`摩擦出非常强大的火花💥。

### CODE REVIEW

**Comment by GrinZero** [comment]
Please do not close this PR. If you encounter any problems, please submit a comment and I will handle it! The high number of active network requests from `RSSHub` can create a powerful spark of friction with `node-network-devtools`.

**Comment by TonyRL** [comment]
The errors are still the same as https://github.com/DIYgod/RSSHub/pull/17280#issuecomment-2438131085 shown. Please make sure it works, at least on GitHub Codespaces, before sending a PR.

**Comment by GrinZero** [comment]
I have completed the new version and tried it on GitHub Codespaces. This is the current phenomenon:
image
As I mentioned earlier, the error prompt of the latter is a double running defect from 'tsx' and can be ignored. And the former, on the other hand, dealt with the situation where I couldn't successfully open Chrome. Now, when I can't open the browser, it will throw a warning instead of an error.

**Comment by GrinZero** [comment]
image
I am pleased to announce that after several days of fixing, I have found the best debugging method in RSSHub. Now accessing 'node network devtools' will only trigger a single warning and significantly increase support for hot updates.

**Comment by TonyRL** [comment]
One last thing, request headers are empty.

**Comment by GrinZero** [comment]
let me cc

**Comment by GrinZero** [comment]
Thank you for providing the information. I have successfully located and resolved the issue. This is because I overlooked the handling of data that cannot be serialized when dealing with the fetch API. I have submitted a new commit and the issue has been fixed.

**Comment by TonyRL** [comment]
Headers are now showing but incomplete. All requests from RSSHub should have ua, referer and accept header.
https://github.com/DIYgod/RSSHub/blob/afcea8f4ee05bf2f816b16425bec18c8123e3822/lib/utils/request-rewriter/fetch.ts#L23-L41

image

**Comment by GrinZero** [comment]
After a period of trying, I have identified the root cause. https://github.com/GrinZero/node-network-devtools/issues/28

This is really bad news, that is, due to the packaging of `GlobalThis ['fetch']` by `RSSHub`, and the fact that it ultimately calls the `fetch` of the unsupported `undici` library instead of `originFetch`, I am unable to intercept the `headers` within it.

I didn't expect there to be any other request methods besides `ofetch&got`.
I will try to provide support later, but it may take more time.

**Comment by GrinZero** [comment]
I have an idea. The current way that RSShub overwrites global fetch with undici is really rare, and even if I had undici support calls, it would result in repeated display requests. So, I plan to expose some hooks and allow external modification of some data, which will add some code, but the scalability is better, and I will finish it this weekend

File: lib/utils/request-rewriter/fetch.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { getCurrentCell, setCurrentCell } from 'node-network-devtools';
```
Comment (TonyRL) [inline lib/utils/request-rewriter/fetch.spec.ts:1]: Please rename it to `fetch.test.ts`

File: package.json [hunk 2]
```diff
@@ -185,6 +185,7 @@
         "lint-staged": "15.2.10",
         "mockdate": "3.0.5",
         "msw": "2.4.3",
+        "node-network-devtools": "^1.0.22",
```
Comment (TonyRL) [inline package.json:188]: Please pin the dep to its exact version and rebuild pnpm-lock

File: lib/utils/request-rewriter/fetch.spec.ts (hunk 1 above)
Comment (GrinZero) [reply lib/utils/request-rewriter/fetch.spec.ts]: fixed

File: package.json (hunk 2 above)
Comment (GrinZero) [reply package.json]: fixed

**Comment by GrinZero** [comment]
Thank you very much for every communication before the merger. This PR was very meaningful to me and helped me break through some of the limitations of my thinking

**Pull request merged**